crates/astria-composer/src/config.rs

```diff
@@ -0,0 +1,95 @@
+use std::net::SocketAddr;
+
+use figment::{
+    providers::Env,
+    Figment,
+};
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+/// Utility function to read the application's config in one go.
+///
+/// This includes parsing the command line args, getting all environment variables.
+/// This function will short circuit binary execution when `--help` or `--version`
+/// is provided, or if the command line arguments could not be read.
+///
+/// # Errors
+///
+/// An error is returned if the config could not be read.
+pub fn get() -> Result<Config, figment::Error> {
+    Config::from_environment()
+}
+
+/// The high-level config for creating an astria-composer service.
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[serde(deny_unknown_fields)]
+pub struct Config {
+    /// Log level. One of debug, info, warn, or error
+    pub log: String,
+
+    /// Address of the API server
+    pub api_listen_addr: SocketAddr,
+
+    /// Address of the RPC server for the sequencer chain
+    pub sequencer_url: String,
+
+    /// Chain ID that we want to connect to
+    pub chain_id: String,
+
+    /// Address of the RPC server for execution
+    pub execution_url: String,
+}
+
+impl Config {
+    /// Constructs [`Config`] with command line arguments.
+    ///
+    /// The command line arguments have to be explicitly passed in to make
+    /// the config logic testable. [`Config::with_cli`] is kept private because
+    /// the `[config::get]` utility function is the main entry point
+    fn from_environment() -> Result<Config, figment::Error> {
+        let rust_log = Env::prefixed("RUST_").split("_").only(&["log"]);
+
+        Figment::new()
+            .merge(rust_log)
+            .merge(Env::prefixed("ASTRIA_COMPOSER_"))
+            .extract()
```

Comment:
What will happen if all configuration isn't set here?

Can we make some sort of graceful "yo you need to set your environment variables" output. Not necessary here, but would be good to establish as a general pattern.

Reply:
So we are a bit restricted by what `figment` and in turn `serde` provides. Because the config is typed and`serde` is short circuiting and returning an error on the first missing field, that's all that `figment` gets to report. I have changed `main.rs` a bit to not dump the entire error struct at us, but we only get smth like this right now:

```
 failed to read configuration: missing field `log`
```

Unless I can find a better crate for this, we will probably need to implement this ourselves.